lib/src/Uncertainty/Bayesian/RandomWalkMetropolisHastings.cxx

```diff
@@ -122,11 +125,22 @@ void RandomWalkMetropolisHastings::setProposal(const Distribution & proposal)
   if (proposal.getDimension() != marginalIndices_.getSize())
     throw InvalidArgumentException(HERE) << "The proposal density dimension (" << proposal.getDimension()
                                          << ") does not match the block size (" << marginalIndices_.getSize() << ")";
-  if (!(proposal.getSkewness().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon")))
-    throw InvalidArgumentException(HERE) << "The proposal density is not symmetric.";
-  if (!(proposal.getMean().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon")))
-    throw InvalidArgumentException(HERE) << "The proposal density must have a null mean.";
   proposal_ = proposal;
+
+  // In the following, we try do determine whether the proposal distribution is symmetric.
+  // If the proposal distribution has an independent copula, we test a sufficient and necessary condition.
+  // Otherwise, we test a condition that is only sufficient.
+  isProposalSymmetric_ = proposal.getMean().norm() < ResourceMap::GetAsScalar("Distribution-DefaultQuantileEpsilon"); // necessary condition for symmetry
+  if (proposal.hasIndependentCopula())
+  {
+    for (UnsignedInteger j = 0; j < proposal.getDimension(); ++ j)
+      isProposalSymmetric_ = isProposalSymmetric_  && proposal.getMarginal(j).isElliptical(); // zero-mean independent marginals are elliptical
```

Comment:
maybe a quick return here:
```
for (UnsignedInteger j = 0; j < proposal.getDimension(); ++ j)
{
  if (!isProposalSymmetric_) break;
  isProposalSymmetric_ = isProposalSymmetric_  && proposal.getMarginal(j).isElliptical(); // zero-mean independent marginals are elliptical
}
```
BTW comments are preferred on top of the line excepted for very short comments as for closing braces otherwise they need horizontal scrolling to be read

Reply:
OK, fixed.